sdks/java/extensions/sorter/src/main/java/org/apache/beam/sdk/extensions/sorter/ExternalSorter.java

```diff
@@ -0,0 +1,237 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.beam.sdk.extensions.sorter;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+import java.util.UUID;
+import org.apache.beam.sdk.values.KV;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.io.BytesWritable;
+import org.apache.hadoop.io.SequenceFile;
+import org.apache.hadoop.io.SequenceFile.CompressionType;
+import org.apache.hadoop.io.SequenceFile.Sorter.RawKeyValueIterator;
+import org.apache.hadoop.io.SequenceFile.Writer;
+import org.apache.hadoop.mapred.JobConf;
+
+/** Does an external sort of the provided values using Hadoop's {@link SequenceFile}. */
+public class ExternalSorter implements Sorter {
+  private Options options;
+
+  /** Whether {@code sort()} was already called. */
+  private boolean sortCalled = false;
+
+  /** SequenceFile Writer for writing all input data to a file. */
+  private Writer writer;
+
+  /** Sorter used to sort the input file. */
+  private SequenceFile.Sorter sorter;
+
+  /** Temporary directory for input and intermediate files. */
+  private Path tempDir;
+
+  /** The list of input files to be sorted. */
+  private Path[] paths;
+
+  private boolean initialized = false;
+
+  /** {@code Options} contains configuration of the sorter. */
+  public static class Options implements Serializable {
+    private String tempLocation = "/tmp";
+    private int memoryMB = 100;
+
+    /** Sets the path to a temporary location where the sorter writes intermediate files. */
+    public void setTempLocation(String tempLocation) {
+      if (tempLocation.startsWith("gs://")) {
+        throw new IllegalArgumentException("Sorter doesn't support GCS temporary location.");
+      }
+
+      this.tempLocation = tempLocation;
+    }
+
+    /** Returns the configured temporary location. */
+    public String getTempLocation() {
+      return tempLocation;
+    }
+
+    /** Sets the size of the memory buffer in megabytes. */
+    public void setMemoryMB(int memoryMB) {
+      this.memoryMB = memoryMB;
+    }
+
+    /** Returns the configured size of the memory buffer. */
+    public int getMemoryMB() {
+      return memoryMB;
+    }
+  }
+
+  /** Returns a {@code Sorter} configured with the given {@code options}. */
+  public static ExternalSorter create(Options options) {
+    return new ExternalSorter(options);
+  }
+
+  /**
+   * Adds a given record to the sorter.
+   *
+   * <p>Records can only be added before calling {@code sort()}.
+   */
+  @Override
```

Comment:
drop Javadoc on overridden methods? I do not think you are changing the text from the superclass.